src/jit/compiler.h

```diff
@@ -2444,6 +2444,9 @@ class Compiler
         DNER_DepField,    // It is a field of a dependently promoted struct
         DNER_NoRegVars,   // opts.compFlags & CLFLG_REGVAR is not set
         DNER_MinOptsGC,   // It is a GC Ref and we are compiling MinOpts
+#if !defined(LEGACY_BACKEND) && defined(_TARGET_X86_)
+        DNER_LongParamField, // It is a decomposed field of a long parameter.
```

Comment:
Actually, it affects both. We don't support enregistering promoted "aggregate" parameters with more than one field. I've fixed this.